invenio_rdm_migrator/streams/actions/drafts.py

```diff
@@ -0,0 +1,234 @@
+
+    name = "create-draft"
+
+    def __init__(
+        self,
+        tx_id,
+        pid,
+        bucket,
+        draft,
+        parent,
+        parents_state,
+        records_state,
+        communities_state,
+        pids_state,
+    ):
+        """Constructor."""
+        super().__init__(
+            tx_id,
+            pks=[
+                ("pid", "id", generate_pk),
+                ("draft", "id", generate_uuid),
+                ("parent", "id", generate_uuid),
+                ("parent", "json.pid", generate_recid),
+            ],
+        )
+        assert pid and bucket and draft and parent  # i.e. not None as parameter
+        self.pid = pid
+        self.bucket = bucket
+        self.draft = draft
+        self.parent = parent
+        # state related
+        self.parents_state = parents_state
+        self.records_state = records_state
+        self.communities_state = communities_state
+        self.pids_state = pids_state
+
+    def _generate_rows(self, **kwargs):
+        """Generates rows for a new draft."""
+        yield from self._generate_pid_rows(**kwargs)
+        yield from self._generate_bucket_rows(**kwargs)
+        yield from self._generate_draft_rows(**kwargs)
+
+    def _generate_pid_rows(self, **kwargs):
+        """Generates rows for a new draft."""
+        if self.pid["pid_type"] != "depid":
+            # FIXME: temporary fix to test the microseconds conversion
+            # should be moved to a mixin or similar on the transform step
+            from datetime import datetime
+
+            self.pid["created"] = datetime.fromtimestamp(
+                self.pid["created"] / 1_000_000
+            )
+            self.pid["updated"] = datetime.fromtimestamp(
+                self.pid["updated"] / 1_000_000
+            )
+
+            # note would raise an exception if it exists
+            self.pids_state.add(
+                self.pid["pid_value"],  # recid
+                {
+                    "id": self.pid["id"],
+                    "pid_type": self.pid["pid_type"],
+                    "status": self.pid["status"],
+                    "obj_type": self.pid["object_type"],
+                    "created": self.pid["created"],
+                },
+            )
+            yield Operation(OperationType.INSERT, PersistentIdentifier(**self.pid))
+
+    def _generate_bucket_rows(self, **kwargs):
+        """Generates rows for a new draft."""
+        # FIXME: temporary fix to test the microseconds conversion
+        # should be moved to a mixin or similar on the transform step
+        from datetime import datetime
+
+        self.bucket["created"] = datetime.fromtimestamp(
+            self.bucket["created"] / 1_000_000
+        )
+        self.bucket["updated"] = datetime.fromtimestamp(
+            self.bucket["updated"] / 1_000_000
+        )
+
+        yield Operation(OperationType.INSERT, FilesBucket(**self.bucket))
+
+    def _generate_draft_rows(self, **kwargs):
```

Comment:
note for reviewers: this is a PoC code. It does not take into account things like:
- File additions
- DOIs
- Publishing